Contents/Libraries/Shared/subliminal_patch/providers/supersubtitles.py

```diff
@@ -0,0 +1,400 @@
+        # format
+        if video.format and self.version and video.format.lower() in self.version.lower():
+            matches.add('format')
+        # other properties
+        # matches |= guess_matches(video, guessit(self.release_info.encode("utf-8")))
+
+        self.matches = matches
+        return matches
+
+
+class SuperSubtitlesProvider(Provider, ProviderSubtitleArchiveMixin):
+    """SuperSubtitles Provider."""
+    languages = {Language('hun', 'HU')} | {Language(l) for l in [
+        'hun', 'eng'
+    ]}
+    video_types = (Episode, Movie)
+    # https://www.feliratok.info/?search=&soriSorszam=&nyelv=&sorozatnev=The+Flash+%282014%29&sid=3212&complexsearch=true&knyelv=0&evad=4&epizod1=1&cimke=0&minoseg=0&rlsr=0&tab=all
+    server_url = 'https://www.feliratok.info/'
+    subtitle_class = SuperSubtitlesSubtitle
+    hearing_impaired_verifiable = False
+    multi_result_throttle = 2  # seconds
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers = {'User-Agent': os.environ.get("SZ_USER_AGENT", "Sub-Zero/2")}
+
+    def terminate(self):
+        self.session.close()
+
+    def get_language(self, text):
+        if text == 'Magyar':
+            return Language.fromsupersubtitles('hu')
+        if text == 'Angol':
+            return Language.fromsupersubtitles('en')
+        return None
+
+    def find_imdb_id(self, sub_id):
+        """
+
+        """
+
+        url = self.server_url + "index.php?tipus=adatlap&azon=a_" + sub_id
+        # url = https://www.feliratok.info/index.php?tipus=adatlap&azon=a_1518600916
+        logger.info('Get IMDB id from URL %s', url)
+        r = self.session.get(url, timeout=10).content
+
+        soup = ParserBeautifulSoup(r, ['lxml'])
+        links = soup.find_all("a")
+
+        for value in links:
+            if "imdb.com" in str(value):
+                # <a alt="iMDB" href="http://www.imdb.com/title/tt2357547/" target="_blank"><img alt="iMDB" src="img/adatlap/imdb.png"/></a>
+                imdb_id = re.findall(r"(?<=www.imdb.com\/title\/).*(?=\/\")", str(value))[0]
+                return imdb_id
+
+        return None
+
+    def find_id(self, series, year, original_title):
+        """
+        We need to find the id of the series at the following url:
+        https://www.feliratok.info/index.php?term=SERIESNAME&nyelv=0&action=autoname
+        Where SERIESNAME is a searchable string.
+        The result will be something like this:
+        [{"name":"DC\u2019s Legends of Tomorrow (2016)","ID":"3725"},{"name":"Miles from Tomorrowland (2015)","ID":"3789"}
+        ,{"name":"No Tomorrow (2016)","ID":"4179"}]
+
+        """
+
+        # Search for exact name
+        url = self.server_url + "index.php?term=" + series + "&nyelv=0&action=autoname"
+        # url = self.server_url + "index.php?term=" + "fla"+ "&nyelv=0&action=autoname"
+        logger.info('Get series id from URL %s', url)
+        r = self.session.get(url, timeout=10).content
```

Comment:
If `r` is something like `[{"name":"DC\u2019s Legends of Tomorrow (2016)","ID":"3725"},{"name":"Miles from Tomorrowland (2015)","ID":"3789"}]`, why don't you just use r.json() to get a usable python object, instead of manually parsing it?

Reply:
I use the manual parsing because:
When I try to parse it lik this:
results = r.json()
logger.info('loged_result %s', result['name'])

instead of:
DC\u2019s Legends of Tomorrow (2016)
or 
DC's Legends of Tomorrow (2016)

I get this: 
loged_result DCâ€™s Legends of Tomorrow (2016)

Probably some decode encode issue, but i dont find the solution. Any hint?

Reply:
I founded the problem:
I was replacing ' char but with changed coding + json this char become to this ’